syllabus/curricula/plr/task_sampler.py

```diff
@@ -261,12 +275,76 @@ def _sample_unseen_level(self):
 
         return task
 
+    def compute_returns(self, gamma, gae_lambda, rewards, value_preds, masks):
+        assert self.requires_value_buffers, "Selected strategy does not use compute_rewards."
+        gae = 0
+        returns = torch.zeros_like(rewards)
+        for step in reversed(range(rewards.size(0))):
+            delta = (
+                    rewards[step]
+                    + gamma * value_preds[step + 1] * masks[step + 1]
+                    - value_preds[step]
+            )
+            gae = delta + gamma * gae_lambda * masks[step + 1] * gae
+            returns[step] = gae + value_preds[step]
+        return returns
+
+    def evaluate_task(self, task, env, action_value_fn, gamma, gae_lambda):
+        if env is None:
+            raise ValueError("Environment object is None. Please ensure it is properly initialized.")
+
+        obs = env.reset(next_task=task)
+        done = False
+        rewards = []
+        masks = []
+        values = []
+
+        while not done:
+            action, value = action_value_fn(obs)
+            obs, rew, term, trunc, info = env.step(action)
+
+            rewards.append(rew)
+            masks.append(not (term or trunc))
+            values.append(value)
+
+            # Check if the episode is done
+            if term or trunc:
+                done = True
+
+        # Compute returns after the episode is complete
+        returns = self.compute_returns(gamma, gae_lambda, rewards, values, masks)
+
+        return {
+            "tasks": task,
+            "masks": masks,
+            "rewards": rewards,
+            "value_preds": values,
+            "returns": returns
+        }
```

Comment:
Once you clean up everything else, we should see how fast this is and come up with some ideas to optimize it